pycaret/internal/plots/time_series.py

```diff
@@ -791,3 +832,106 @@ def plot_predictions_with_confidence(
         return fig, data, predictions, upper_interval, lower_interval
     else:
         return fig
+
+
+def plot_time_series_decomposition(
+    data: pd.Series,
+    trend_component:pd.Series,
+    seasonl_component:pd.Series,
+    resdiual_component:pd.Series,
+    model_name: Optional[str] = None,
+    return_data: bool = False,
+    show: bool = True,
+    data_kwargs: Dict = None,
+    fig_kwargs: Dict = None,
+):
+
+    if data_kwargs is None:
+        data_kwargs = {}
+    if fig_kwargs is None:
+        fig_kwargs = {}
+
+    title = (
+        "Time Series Decomposition"
+        if data.name is None
+        else f"Time Series Decomposition | {data.name}"
+    )
+
+    fig = make_subplots(
+        rows=4,
+        cols=1,
+        row_heights=[
+            0.25,
+            0.25,
+            0.25,
+            0.25,
+        ],
+
+        row_titles=["Actual", "Seasonal", "Trend", "Residual"],
+    )
+
+    fig.add_trace(
+        go.Scatter(
+            x=data.index.to_timestamp(),
+            y=data.observed,
+            line=dict(color="#1f77b4", width=2),
+            mode="lines+markers",
+            name="Actual",
+            marker=dict(size=2, ),
+        ),
+
+        row=1,
+        col=1,
+    )
+
+    fig.add_trace(
+        go.Scatter(
+            x=data.index.to_timestamp(),
+            y=seasonl_component,
+            line=dict(color="#1f77b4", width=2),
+            mode="lines+markers",
+            name="Seasonal",
+            marker=dict(size=2, ),
+        ),
+        row=2,
+        col=1,
+    )
+
+    fig.add_trace(
+        go.Scatter(
+            x=data.index.to_timestamp(),
+            y=trend_component,
+            line=dict(color="#1f77b4", width=2),
+            mode="lines+markers",
+            name="Trend",
+            marker=dict(size=2, ),
+        ),
+        row=3,
+        col=1,
+    )
+
+    fig.add_trace(
+        go.Scatter(
+            x=data.index.to_timestamp(),
+            y=resdiual_component,
+            line=dict(color="#1f77b4", width=2),
+            mode="markers",
+            name="Resdiuals",
+            marker=dict(size=4, ),
+        ),
+        row=4,
+        col=1,
+    )
+    fig.update_layout(title=title)
```

Comment:
Could you add an option to change the fig size here as you have done in other functions?